src/components/app/data/hooks/useEnterpriseCustomer.test.jsx

```diff
@@ -1,53 +1,111 @@
-  enterpriseCustomer: mockEnterpriseCustomer,
-  enterpriseCustomerUserRoleAssignments: [],
-  activeEnterpriseCustomer: null,
-  activeEnterpriseCustomerUserRoleAssignments: [],
+  transformed: {
+    enterpriseCustomer: mockEnterpriseCustomer,
+    enterpriseCustomerUserRoleAssignments: [],
+    activeEnterpriseCustomer: null,
+    activeEnterpriseCustomerUserRoleAssignments: [],
+    allLinkedEnterpriseCustomerUsers: [],
+    staffEnterpriseCustomer: null,
+    enterpriseFeatures: {
+      isBFFEnabled: false,
+    },
+    shouldUpdateActiveEnterpriseCustomerUser: false,
+  },
+};
+const mockBFFDashboardData = {
+  enterpriseCustomer: {
+    ...mockEnterpriseCustomer,
+    isBFFEnabled: true,
+  },
   allLinkedEnterpriseCustomerUsers: [],
-  enterpriseFeatures: {},
-  staffEnterpriseCustomer: null,
+  enterpriseFeatures: {
+    isBFFEnabled: true,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+  enterpriseCustomerUserSubsidies: {
+    subscriptions: {
+      customerAgreement: {},
+      subscriptionLicenses: [],
+      subscriptionLicensesByStatus: {
+        activated: [],
+        assigned: [],
+        expired: [],
+        revoked: [],
+      },
+    },
+  },
+  enterpriseCourseEnrollments: [],
+  errors: [],
+  warnings: [],
 };
 
+const mockExpectedEnterpriseCustomers = (isMatchedRoute) => (isMatchedRoute
+  ? mockBFFDashboardData.enterpriseCustomer
+  : mockEnterpriseLearnerData.transformed.enterpriseCustomer);
+
 describe('useEnterpriseCustomer', () => {
-  const Wrapper = ({ children }) => (
+  const Wrapper = ({ routes = null, children }) => (
     <QueryClientProvider client={queryClient()}>
-      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
-        {children}
-      </AppContext.Provider>
+      <MemoryRouter initialEntries={[routes]}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
     </QueryClientProvider>
   );
   beforeEach(() => {
     jest.clearAllMocks();
     fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
+    fetchEnterpriseLearnerDashboard.mockResolvedValue(mockBFFDashboardData);
+    useParams.mockReturnValue({ enterpriseSlug: 'test-slug' });
   });
-  it('should return enterprise customer metadata correctly', async () => {
-    const { result, waitForNextUpdate } = renderHook(() => useEnterpriseCustomer(), { wrapper: Wrapper });
+  it.each([
+    { isMatchedRoute: false },
+    { isMatchedRoute: true },
+  ])('should return enterprise customers correctly (%s)', async ({ isMatchedRoute }) => {
+    const mockSelect = jest.fn(data => data.transformed);
+    const { result, waitForNextUpdate } = renderHook(
+      () => {
+        if (isMatchedRoute) {
+          return useEnterpriseCustomer({ select: mockSelect });
+        }
+        return useEnterpriseCustomer();
+      },
+      {
+        wrapper: ({ children }) => Wrapper({
```

Comment:
Similar feedback around rendering `Wrapper` with JSX vs. calling it like a function.